### DISCUSSION TITLE
Dotnet Host SDK/PDK - Unable to create Extism plugin: unknown import: `wasi_snapshot_preview1::sock_accept` has not been defined

### DISCUSSION THREAD

**Discussion by Simonl9l** [extism/extism#689] [Q&A]
Environment:
- Host - DotNet SDK v1.0.1 - Runtime v1.1.0
- Plugin DotNnet PDK v1.0.1
- Using Host Functions

Given a defined `manifest` and `hostFunctions[]` and calling:
```C#
_plugin = new Plugin(manifest, _hostFunctions, withWasi: false);
``` 
I get the error:
```
Unable to create Extism plugin: unknown import: `wasi_snapshot_preview1::sock_accept` has not been defined
```

if I set the `withWasi: true` I get the error:
```
Unable to create Extism plugin: No such file or directory (os error 2)
```

I have confirmed the path to the `.wasm` file is valid.

Thoughts?

**Comment by mhmd-azeez** [comment]
Hi,

> Unable to create Extism plugin: unknown import: `wasi_snapshot_preview1::sock_accept` has not been defined

 .NET plugins require WASI (so `withWasi` has to be `true`).

> Unable to create Extism plugin: No such file or directory (os error 2)

This means there is a problem with the path of the `.wasm` file specified in the manifest. Can you try giving it the absolute path? Please note that the working dir of `dotnet run` is different (project dir) from the working dir of VS F5 (bin/debug)

P.S.: Please take a look at this example for accessing the file system:
https://github.com/extism/dotnet-sdk/blob/1b647f0f45775dd94cfe51771f807f900a5faa73/test/Extism.Sdk/BasicTests.cs#L83

This is the text file:
https://github.com/extism/dotnet-sdk/blob/1b647f0f45775dd94cfe51771f807f900a5faa73/test/Extism.Sdk/data/test.txt
